src/com/facebook/buck/apple/AppleDescriptions.java

```diff
@@ -207,7 +211,18 @@ public static void populateCxxConstructorArg(
                     arg))
             .build();
 
-    output.srcs = arg.srcs;
+    if (arg.srcs.isPresent()) {
```

Comment:
One of the downside of this approach (modify the action graph during transformation phrase) is that you have to filter the source codes for swift & non-swift in apple targets. Doing it in the parsing phrase makes more sense to me. Any idea for improving this?